### PR TITLE
Ocaml Compatibility, Strings Deprecated Warning

### PR DESCRIPTION
**Pull request by rijeshaugustine**
This should resolve issue #1703 deprecated strings warning.

### CODE REVIEW

File: sw/lib/ocaml/compat.ml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  BYTES.contains c s
+
+let bytes_length = fun len ->
+  BYTES.length len
+
+let bytes_make = fun n c->
+  BYTES.make n c
+
+let bytes_copy = fun s->
+	BYTES.copy s
```
Comment (gautierhattenberger) [inline sw/lib/ocaml/compat.ml]: please keep the same indentation for all the file

**Comment by gautierhattenberger** [comment]
The `open` instruction is a kind of `using namespace` in C++, it allows to use functions without specifying the module. But here you are using the `Compat.somthing` (and it is the correct way to go) so it should not be necessary.

**Comment by flixr** [comment]
There are still some warnings, e.g.
```
OC http.ml
File "http.ml", line 24, characters 4-44:
Warning 5: this function application is partial,
maybe some arguments are missing.
```

@gautierhattenberger I leave this to you as you have way more OCaml experience and knowledge...
If you think this is a good solution: merge it

**Comment by rijeshaugustine** [comment]
I don't think the warnings are due to strings being deprecated. I was unclear as to how exactly this should be handled. 

The error is due to:
`pipeline # set_proxy_from_environment ();`

method set_proxy_from_environment : insecure:bool -> unit -> unit
Inspect the environment variables http_proxy and no_proxy and set the proxy options from them. If insecure is set, basic authentication is enabled (otherwise only secure auth methods like Digest).

I can look at it further.

**Comment by gautierhattenberger** [comment]
Since Ocamlnet 4.0.4, you need to specify the `insecure` input value. Probably something like:
```
pipeline # set_proxy_from_environment ~insecure:false ();
```
An other candidate for the Compat module...

**Pull request merged**